theia-extensions/cdt-cloud-pico/src/browser/project-editor/project-editor-widget.tsx

```diff
@@ -0,0 +1,226 @@
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { CommandService, nls } from '@theia/core';
+import { codicon, LabelProvider, NavigatableWidget, NavigatableWidgetOptions, ReactWidget } from '@theia/core/lib/browser';
+import { BinaryBuffer } from '@theia/core/lib/common/buffer';
+import URI from '@theia/core/lib/common/uri';
+import {
+    inject,
+    injectable,
+    postConstruct
+} from '@theia/core/shared/inversify';
+import * as React from '@theia/core/shared/react';
+import { FileService } from '@theia/filesystem/lib/browser/file-service';
+import { ProjectCommands } from '../project-command-contribution';
+
+export interface ProjectEditorWidgetOptions extends NavigatableWidgetOptions {
+    id: string;
+}
+export const ProjectEditorWidgetOptions = Symbol('ProjectEditorWidgetOptions');
+
+export interface ProjectSettings {
+    hardware: string;
+    serialPrinting: string,
+    name?: string,
+    url?: string
+}
+
+const DEFAULT_PROJECT_SETTINGS: ProjectSettings = {
+    hardware: 'pico',
+    serialPrinting: 'USB',
+};
+
+@injectable()
+export class ProjectEditorWidget extends ReactWidget implements NavigatableWidget {
+
+    @inject(ProjectEditorWidgetOptions) protected options: ProjectEditorWidgetOptions;
+    @inject(LabelProvider) protected labelProvider: LabelProvider;
+    @inject(FileService) protected fileService: FileService;
+    @inject(CommandService) protected commandService: CommandService;
+
+    protected projectSettingsUri: URI;
+    protected projectSettings?: ProjectSettings;
+
+    getResourceUri(): URI | undefined {
+        return new URI(this.options.uri);
+    }
+
+    createMoveToUri(resourceUri: URI): URI | undefined {
+        return undefined;
+    }
+
+    @postConstruct()
+    init(): void {
+        this.id = this.options.id;
+        this.title.label = this.getProjectName();
+        this.title.iconClass = this.getProjectIconClass();
+        this.title.closable = true;
+        this.projectSettingsUri = new URI(this.options.uri);
+        this.readProjectSettings().then(projectSettings => {
+            this.projectSettings = projectSettings;
+            this.update();
+        });
+    }
+
+    protected getProjectName(): string {
+        const uri = this.getResourceUri();
+        const parent = uri?.parent;
+        if (!parent) {
+            return nls.localize('cdt-cloud-pico/project-editor/unknown', 'Unknown');
+        }
+        return this.labelProvider.getName(parent);
+    }
+
+    protected getProjectIconClass(): string {
```

Comment:
It would be cool if the editor also uses the cdt cloud icon as the text editor does:
![image](https://user-images.githubusercontent.com/47414588/203772572-6c7ab859-f2ea-4323-ae4c-62c90457f12f.png)

Reply:
I'll do that in a follow-up change, thanks!